### PR TITLE
Fix calculation of cloud cover percentage

### PR DESCRIPTION
**Pull request by wlatanowicz**

### CODE REVIEW

File: pyastroweatherio/dataclasses.py [hunk 1]
```diff
@@ -71,8 +75,12 @@ def seeing(self) -> int:
...
     @property
     def seeing_percentage(self) -> int:
-        """Return Seeing."""
-        return int((100 + 100 / 7 - self._seeing * 100 / 7))
+        """Return Seeing.
+        Plain value is an int between 1 and 8 (1 is below .5", 8 is above 2.5"; 1 is best;
+        see https://github.com/Yeqzids/7timer-issues/wiki/Wiki#astro),
+        converting it to value between 0 and 100.
+        """
+        return int(100 - 100 * (self._seeing - 1 ) / 7)
```
Comment (wlatanowicz) [inline pyastroweatherio/dataclasses.py:83]: Mathematically it's the same, but I find my version easier to read. I can revert it if you do not agree.

File: pyastroweatherio/dataclasses.py [hunk 2]
```diff
@@ -81,8 +89,12 @@ def transparency(self) -> int:
...
     @property
     def transparency_percentage(self) -> int:
-        """Return Transparency."""
-        return int((100 + 100 / 7 - self._transparency * 100 / 7))
+        """Return Transparency.
+        Plain value is an int between 1 and 8 (1 is below 0.3, 8 is above 1; 1 is best;
+        see https://github.com/Yeqzids/7timer-issues/wiki/Wiki#astro),
+        converting it to value between 0 and 100.
+        """
+        return int(100 - 100 * (self._transparency - 1 ) / 7)
```
Comment (wlatanowicz) [inline pyastroweatherio/dataclasses.py:97]: Same as above.

File: pyastroweatherio/dataclasses.py [hunk 3]
```diff
@@ -61,8 +61,12 @@ def cloudcover(self) -> int:
...
     @property
     def cloudcover_percentage(self) -> int:
-        """Return Cloud Coverage."""
-        return int((100 + 100 / 8 - self._cloudcover * 100 / 8))
+        """Return Cloud Coverage.
+        Plain value is an int between 1 and 9 (1 is 0-6%, 9 is 94-100%;
+        see https://github.com/Yeqzids/7timer-issues/wiki/Wiki#astro),
+        converting it to value between 0 and 100.
+        """
+        return int(100 * (self._cloudcover - 1 ) / 8)
```
Comment (wlatanowicz) [inline pyastroweatherio/dataclasses.py:69]: The original version inverted the scale.

**Comment by mawinkler** [comment]
In regards to the cloud coverage, actually, it works as designed.

Here’s a quote from the Readme of the Lovelace card:

"The percentages that are calculated for the clouds, seeing and transparency are to be read like 100% is perfect, 0% is bad. This also means that a percentage of e.g. 87% for the cloud does stand for a sky being nearly cloud-free ;-).“

I know this might look confusing. My intention was to have the same „measurement“ for the three parameters currently influencing the condition. So they needed all three to be in the range of 0% to 100%. In other words, it’s a condition view.

Cheers,
Markus

**Pull request closed without merging**